src/cosalib/qemuvariants.py

```diff
@@ -304,24 +304,31 @@ def mutate_image(self):
             log.info(f"Moving {work_img} to {final_img}")
             shutil.move(work_img, final_img)
 
-        if self.gzip:
-            sha256 = sha256sum_file(final_img)
-            size = os.stat(final_img).st_size
-            temp_path = f"{final_img}.tmp"
-            with open(temp_path, "wb") as fh:
-                runcmd(['gzip', '-9c', final_img], stdout=fh)
-            shutil.move(temp_path, final_img)
-            meta_patch.update({
-                'skip-compression': True,
-                'uncompressed-sha256': sha256,
-                'uncompressed-size': size,
-            })
-
-        if self.skip_compression:
-            meta_patch.update({
-                'skip-compression': True
-            })
-
+        if self.compression is not None:
+            if self.compression == "skip":
+                meta_patch.update({
+                    'skip-compression': True
+                })
+            else:
+                sha256 = sha256sum_file(final_img)
+                size = os.stat(final_img).st_size
+                temp_path = f"{final_img}.tmp"
+
+                match self.compression:
+                    case "gzip":
+                        rc = ['gzip', '-9c', final_img]
+                    case "zip":
+                        rc = ['zip', '-9', final_img + ".zip", final_img]
```

Comment:
The hardcoded suffix isn't right.  `image_suffix` should be `vhdx.zip`, and `zip` should write to stdout so the atomic rename below works correctly.

Reply:
Looks like you haven't updated `image_suffix`.  Also, `zip` doesn't use `-c` for writing to stdout.  I believe you'll want `['zip', '-9', '-'. final_img]`.